### ISSUE TITLE
[hydra] allow auth for all /admin apis

### ISSUE THREAD

**Issue by TuxCoder** [ory/sdk#18]
When I read the api specs correctly currently auth is only possible for a hand of admin APIs.

The whole /admin api should not be public but it is not possible to us oauth or basic auth for the login challange and the consent challange.

Did I understand something wrong or is this a bug in the specs

Currently my python client does not send the set username/password as base-auth.
The server runs hydra with an nginx in front for doing basic auth of the api.

**Comment by TuxCoder** [comment]
after a bit of a setup time, I could generate a new client.
Now it works! :)
See example patch https://github.com/TuxCoder/sdk/commit/cb440f8d1b84453d1e1b5296df6b6370212ab8ea

I added only the auth/security lines, no additional repsone type for a 401, so the patch is not complete and not very nice, more like a POC.
Also not sure where this should be patches, as this is not realy part of the hydra api, but will be a very common used method IMHO, as you need to have some restriction to the admin api.

Another option would be to generate a separate client for different frontends (base auth/oauth proxy), or the user need to do it by himself, like I now, but then there should be more documentation how to do it.

**Comment by aeneasr** [comment]
So is the problem that you have protected the Admin API endpoint with e.g. Basic Auth and now want to use the SDK to access that?

**Comment by TuxCoder** [comment]
yes, that would be the plan, thought this SDK is for accessing the API.

But I understand that the API specs is not wrong from the application view.
Only how it will get used normally, in real world application where you need authentication.
I also understand that hydra itself will not deal with that, that is also totally fine for me, for that reason I use nginx as a proxy.

Still I would not know where to patch this, when I understand correctly, this specs are generated from the go code itself, there for sure exist nor authentication for the api.

So my Idea would be to have a patch pipeline for this stuff, to provide a sdk that supports authentication.

Not sure if that makes sense, but I think I will not be the only one that runs into this issue.
Or even worse they don't protect the admin API at all, because otherwise it is not working.

I saw that the new version of hydra supports more unix sockets, what is a partial solution.
*  not supported by python generated code :/ (upstream problem from different stuff)
* when it's not running on the same machine it is not a option.

This problem also exist when using oathkeeper instant of nginx.

**Comment by aeneasr** [comment]
Thank you for your thoughts here! You are correct, maybe we should add it (with a disclaimer) to the core spec. If that were the case, this would happen [here for example](https://github.com/ory/hydra/blob/master/client/handler.go#L77).

Here's an example security definition: https://github.com/ory/hydra/blob/master/oauth2/handler.go#L264
